lib/registry.dart

```diff
@@ -212,9 +212,9 @@ class Registry<T> {
   /// Tags are compared using [Object.==] equality.
   ///
   /// Fails if any of the elements are not in the registry.
-  Future addTags(Iterable<T> elements, Iterable<Object> tags) {
-    List<Object> ids = elements.map(_getId).toList(growable: false);
-    return _addTags(ids, tags);
+  Future addTags(Iterable<T> elements, Iterable<Object?> tags) {
+    List<Object?> ids = elements.map(_getId).toList(growable: false);
```

Comment:
Just `var ids = ...`. The RHS does create a `List<int>`, so there was never any need to up-cast it to `List<Object>`.